app/src/main/java/com/invisibleteam/goinvisible/mvvm/edition/adapter/EditItemAdapter.java

```diff
@@ -1,64 +1,89 @@
 package com.invisibleteam.goinvisible.mvvm.edition.adapter;
 
+import android.content.Context;
 import android.support.annotation.VisibleForTesting;
 import android.support.v7.widget.RecyclerView;
 import android.view.LayoutInflater;
 import android.view.View;
 import android.view.ViewGroup;
 
 import com.invisibleteam.goinvisible.R;
-import com.invisibleteam.goinvisible.databinding.EditItemViewBinding;
+import com.invisibleteam.goinvisible.helper.EditItemAdapterHelper;
 import com.invisibleteam.goinvisible.model.Tag;
 import com.invisibleteam.goinvisible.mvvm.edition.OnTagActionListener;
-import com.invisibleteam.goinvisible.util.TextViewUtil;
 
 import java.util.ArrayList;
 import java.util.List;
 
-class EditItemAdapter extends RecyclerView.Adapter<EditItemAdapter.ViewHolder> {
+import static com.invisibleteam.goinvisible.model.TagGroupType.ADVANCED;
+import static com.invisibleteam.goinvisible.model.TagGroupType.DEVICE_INFO;
+import static com.invisibleteam.goinvisible.model.TagGroupType.IMAGE_INFO;
+import static com.invisibleteam.goinvisible.model.TagGroupType.LOCATION_INFO;
+
+class EditItemAdapter extends RecyclerView.Adapter<ViewHolder> {
+
+    private static final int IMAGE_INFO_SECTION_POSITION = 0;
+    private static final int LOCATION_INFO_SECTION_POSITION = 21;
+    private static final int DEVICE_INFO_SECTION_POSITION = 47;
+    private static final int ADVANCED_INFO_SECTION_POSITION = 59;
 
-    private List<Tag> tagsList;
     private List<Tag> baseTagsList;
     private OnTagActionListener onTagActionListener;
+    private Context context;
+    private EditItemAdapterHelper helper;
 
-    EditItemAdapter() {
-        tagsList = new ArrayList<>();
+    EditItemAdapter(EditItemAdapterHelper helper) {
+        this.helper = helper;
     }
 
     @Override
     public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
-        final View itemView = LayoutInflater.from(parent.getContext()).inflate(R.layout.edit_item_view, null, false);
-        ViewHolder viewHolder = new ViewHolder(itemView);
-        viewHolder.editItemViewBinding.clearButton.setOnClickListener(v -> {
-            if (onTagActionListener != null) {
-                Tag tag = (Tag) itemView.getTag();
-                onTagActionListener.onClear(tag);
-            }
-        });
-        viewHolder.itemView.setOnClickListener(view -> {
-            if (onTagActionListener != null) {
-                Tag tag = (Tag) itemView.getTag();
-                onTagActionListener.onEditStarted(tag);
-            }
-        });
+        context = parent.getContext();
+
+        if (viewType != -1) {
```

Comment:
please make some defalut value, not hardcoded -1, it is used at least in 2 places